cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -1047,5 +1053,49 @@ protected Future<Void> isReady(String namespace, String podName) {
                 return Future.failedFuture(error);
             });
     }
+
+    /**
+     * Checks from the Pod labels if the Kafka node is already a broker or not.
+     *
+     * @param pod   Current Pod
+     *
+     * @return  Optional with true if the pod is already a broker, false if it is not broker or empty optional
+     * if the label is not present.
+     */
+    /* test */ static Optional<Boolean> isAlreadyBroker(Pod pod)    {
+        return checkBooleanLabel(pod, Labels.STRIMZI_BROKER_ROLE_LABEL);
+    }
+
+    /**
+     * Checks from the Pod labels if the Kafka node is already a controller or not.
+     *
+     * @param pod   Current Pod
+     *
+     * @return  Optional with true if the pod is already a controller, false if it is not controller or empty optional
+     * if the label is not present.
+     */
+    /* test */ static Optional<Boolean> isAlreadyController(Pod pod)    {
+        return checkBooleanLabel(pod, Labels.STRIMZI_CONTROLLER_ROLE_LABEL);
+    }
+
+    /**
+     * Generic method to extract a boolean value from Kubernetes resource labels
+     *
+     * @param pod           Kube resource with metadata
+     * @param annotation    Name of the label for which we want to extract the boolean value
+     *
+     * @return  Optional with true if the label is present and is set to `true`, false if it is present and not set to
+     * `true` or empty optional if the label is not present.
+     */
+    private static Optional<Boolean> checkBooleanLabel(HasMetadata pod, String annotation)    {
```

Comment:
we are looking for label, maybe the `annotation` parameter name could be misleading.